src/coprocessor/dag/batch_handler.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use protobuf::{Message, RepeatedField};
+
+use kvproto::coprocessor::Response;
+use tipb::select::{Chunk, SelectResponse};
+
+use super::batch_executor::interface::{
+    BatchExecuteStatistics, BatchExecutor, BatchExecutorContext,
+};
+use super::executor::ExecutorMetrics;
+use crate::coprocessor::dag::expr::EvalWarnings;
+use crate::coprocessor::*;
+
+const BATCH_INITIAL_SIZE: usize = 32;
+const BATCH_MAX_SIZE: usize = 1024;
```

Comment:
Do we need to make it configurable?

Reply:
Not very necessary. 1024 is chosen based some research work. See Figure 10 of http://cidrdb.org/cidr2005/papers/P19.pdf

We may need to do our own research however for a better value (because our structure is not the same as the paper) if we have time XD